browser/net/brave_network_delegate_base.cc

```diff
@@ -40,6 +44,30 @@ content::WebContents* GetWebContentsFromProcessAndFrameId(int render_process_id,
 
 }  // namespace
 
+namespace brave {
+
+void RemoveTrackableSecurityHeadersForThirdParty(
+        URLRequest* request,
+        const net::HttpResponseHeaders* original_response_headers,
+        scoped_refptr<net::HttpResponseHeaders>* override_response_headers) {
+  if (!request || (!original_response_headers &&
+                   !override_response_headers->get())) {
+    return;
+  }
+  if (net::registry_controlled_domains::SameDomainOrHost(
+        request->url(), request->site_for_cookies(),
```

Comment:
Please use `request->top_frame_origin()`

Reply:
Similar to `site_for_cookies`, `top_frame_origin` might not be set every-time. (see: https://cs.chromium.org/chromium/src/net/url_request/url_request.h?type=cs&q=top_frame_origin&sq=package:chromium&g=0&l=290)

I think using `GetTabURLFromRenderFrameInfo` might be better. Let me know what you think?

Reply:
`site_for_cookies` is intentionally blank for nested 3rdparty frames while `top_frame_origin` is not. I think it may be empty for requests that are not related to any frame at all, so we can do nothing in this case.

Basically, the comment about its experimental nature is outdated, since Chromium had a successful experiment based on this value, and it will stay with us